src/dappsExplorer/search/DAppsExplorerScreenSearch.tsx

```diff
@@ -0,0 +1,327 @@
+  const error = useSelector(dappsListErrorSelector)
+  const categories = useSelector(dappsCategoriesAlphabeticalSelector)
+  const dappFavoritesEnabled = useSelector(dappFavoritesEnabledSelector)
+  const dappsMinimalDisclaimerEnabled = useSelector(dappsMinimalDisclaimerEnabledSelector)
+  const dappList = useSelector(dappsListSelector)
+  const dappListWithCategoryNames = useSelector(dappListWithCategoryNamesSelector)
+  const language = useSelector(currentLanguageSelector)
+  const favoriteDappsById = useSelector(favoriteDappIdsSelector)
+
+  // Some state lifted up from all and favorite sections
+  const [searchTerm, setSearchTerm] = React.useState('')
+  const [favoriteResultsEmpty, setFavoriteResultsEmpty] = React.useState(false)
+  const [allResultEmpty, setAllResultEmpty] = React.useState(false)
+
+  const { onSelectDapp, ConfirmOpenDappBottomSheet } = useOpenDapp()
+  const { onFavoriteDapp, DappFavoritedToast } = useDappFavoritedToast(sectionListRef)
+  const { openSheet, DappInfoBottomSheet } = useDappInfoBottomSheet()
+
+  useEffect(() => {
+    dispatch(fetchDappsList())
+    ValoraAnalytics.track(DappExplorerEvents.dapp_screen_open)
+  }, [])
+
+  const allSectionResults: SectionData[] = React.useMemo(() => {
+    const allResultsParsed = parseResultsIntoAll(
+      dappListWithCategoryNames,
+      searchTerm,
+      favoriteDappsById
+    )
+    if (allResultsParsed.length === 0) {
+      setAllResultEmpty(true)
+    } else {
+      setAllResultEmpty(false)
+    }
+    return allResultsParsed
+  }, [dappList, searchTerm, favoriteDappsById])
+
+  const emptyListComponent = React.useMemo(() => {
+    if (allResultEmpty && favoriteResultsEmpty) return null
+    return <NoResultsSearch searchTerm={searchTerm} testID="DAppsExplorerScreenSearch" />
+  }, [allResultEmpty, favoriteResultsEmpty])
+
+  return (
+    <SafeAreaView
+      testID="DAppsExplorerScreenSearch"
+      style={styles.safeAreaContainer}
+      edges={['top']}
+    >
+      <DrawerTopBar
+        rightElement={
+          <TopBarIconButton
+            testID="DAppsExplorerScreenSearch/HelpIcon"
+            icon={<Help color={colors.onboardingGreen} />}
+            onPress={openSheet}
+          />
+        }
+        scrollPosition={scrollPosition}
+      />
+      {ConfirmOpenDappBottomSheet}
+      <>
+        {!loading && error && (
+          <View style={styles.centerContainer}>
+            <Text style={fontStyles.regular}>{t('dappsScreen.errorMessage')}</Text>
+          </View>
+        )}
+        {categories.length && (
+          <AnimatedSectionList
+            refreshControl={
+              <RefreshControl
+                tintColor={colors.greenBrand}
+                colors={[colors.greenBrand]}
+                style={styles.refreshControl}
+                refreshing={loading}
+                onRefresh={() => dispatch(fetchDappsList())}
+              />
+            }
+            // @ts-ignore TODO: resolve type error
```

Comment:
Too bad we didn't address this, now it's being copied in multiple places.

```suggestion
            // TODO: resolve type error
            // @ts-expect-error 
```